src/clm/functions.py

```diff
@@ -267,12 +267,14 @@ def discrete_JSD(generated_dist, original_dist, tol=1e-10):
     return jensenshannon(gen, org)
 
 
-def internal_diversity(fps, sample_size=1e4, summarise=True):
+def internal_diversity(fps, sample_size=1e4, summarise=True, seed=None):
     """
     Calculate the internal diversity, defined as the mean intra-set Tanimoto
     coefficient, between a set of fingerprints. For large sets, calculating the
     entire matrix is prohibitive, so a random set of molecules are sampled.
     """
+    if seed:
```

Comment:
I think you want to say `if seed is None`, but again the `set_seed` function does that anyway, so you can perhaps just call it unconditionally

Reply:
Removed `set_seed` from individual functions and updated the `calculate.csv` file accordingly.